app/assets/javascripts/discourse/app/lib/popup.js

```diff
@@ -0,0 +1,163 @@
+import { iconHTML } from "discourse-common/lib/icon-library";
+import I18n from "I18n";
+import tippy from "tippy.js";
+
+const GLOBAL_POPUPS_KEY = "new_user_tips";
+const POPUP_KEYS = ["first-notification", "topic-timeline"];
+
+const instances = {};
+const queue = [];
+
+// Plugin used to implement actions of the two buttons
+const PopupPlugin = {
+  name: "popup",
+
+  fn(instance) {
+    return {
+      onCreate() {
+        instance.popper
+          .querySelector(".btn-primary")
+          .addEventListener("click", (event) => {
+            const { currentUser, popup } = instance.props;
+            hidePopupForever(currentUser, popup);
+            event.preventDefault();
+          });
+
+        instance.popper
+          .querySelector(".btn-flat")
+          .addEventListener("click", (event) => {
+            const { currentUser } = instance.props;
+            hidePopupForever(currentUser, GLOBAL_POPUPS_KEY);
+            event.preventDefault();
+          });
+      },
+    };
+  },
+};
+
+function getUserOptionKey(popup) {
+  return `skip_${popup.replaceAll("-", "_")}_tips`;
+}
+
+export function showPopup(options) {
+  hidePopup(options.popup);
+
+  if (
+    !options.reference ||
+    !options.currentUser ||
+    options.currentUser.get(getUserOptionKey(options.popup))
+  ) {
+    return;
+  }
+
+  if (Object.keys(instances).length > 0) {
+    return addToQueue(options);
+  }
+
+  instances[options.popup] = tippy(options.reference, {
+    placement: options.placement,
+
+    plugins: [PopupPlugin],
+
+    // Current user is used to keep track of popups.
+    currentUser: options.currentUser,
+
+    // Key used to save state.
+    popup: options.popup,
+
+    // Tippy must be displayed as soon as possible and not be hidden unless
+    // the user clicks on one of the two buttons.
+    showOnCreate: true,
+    hideOnClick: false,
+    trigger: "manual",
+
+    // It must be interactive to make buttons work.
+    interactive: true,
+
+    // The default max width is 350px and that is not enough to fit the
+    // buttons.
+    maxWidth: "none",
+    arrow: iconHTML("tippy-rounded-arrow"),
+
+    // It often happens for the reference element to be rerendered. In this
+    // case, tippy must be rerendered too. Having an animation means that the
+    // animation will replay over and over again.
+    animation: false,
+
+    // The `content` property below is HTML.
+    allowHTML: true,
```

Comment:
I wonder if this is needed. Would be safer to start without allowing HTML here, IMO.

Reply:
Any kind of rich content inside Tippy requires this option. It's not ideal, but I don't know of a better method.

Reply:
It is HTML. It is the `content` property on the following lines.